### PR TITLE
Update Azure Quantum Jobs readme, tests and samples to use Quantum Computing sample

### PR DESCRIPTION
**Pull request by vxfield**
This PR updates the Azure.Quantum.Jobs Readme, Sample and Tests to use a Quantum Computing job instead of the recently deprecated Quantum Inspired Optimization (QIO) job.

The SDK itself has not changed and a new release is not needed.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by vxfield** [comment]
@kinelski, could you help review and merge this PR?

**Comment by vxfield** [comment]
@tg-msft, could you help review this PR?

File: sdk/quantum/Azure.Quantum.Jobs/samples/Program.cs [hunk 1]
```diff
@@ -26,7 +28,7 @@ public static void Main(string[] args)
             var location = "your_location";
             var storageContainerName = "your_container_name";
 
-            var credential = new DefaultAzureCredential(true);
+            TokenCredential credential = new DefaultAzureCredential(true);
```
Comment (AlexanderSher) [inline sdk/quantum/Azure.Quantum.Jobs/samples/Program.cs:31]: Why?

File: sdk/quantum/Azure.Quantum.Jobs/tests/QuantumJobClientLiveTests.cs [hunk 2]
```diff
@@ -96,55 +96,44 @@ public async Task JobLifecycleTest()
...
-                        ContentEncoding = "gzip"
-                    };
-                    var blobUploadOptions = new BlobUploadOptions { HttpHeaders = blobHeaders };
-                    blobClient.Upload(problemStreamToUpload, options: blobUploadOptions);
+                    ContentType = "qir.v1"
+                };
+                var blobUploadOptions = new BlobUploadOptions { HttpHeaders = blobHeaders };
+                using (FileStream qirFileStream = File.OpenRead(qirFilePath))
+                {
+                    blobClient.Upload(qirFileStream, options: blobUploadOptions);
```
[the remark this answers is not in the document]
Comment (vxfield) [reply sdk/quantum/Azure.Quantum.Jobs/tests/QuantumJobClientLiveTests.cs:120]: Yes, I'll use that

File: sdk/quantum/Azure.Quantum.Jobs/samples/Program.cs (hunk 1 above)
Comment (vxfield) [reply sdk/quantum/Azure.Quantum.Jobs/samples/Program.cs:31]: I changed it when manually testing it, but will revert it to `var`.

**Review by AlexanderSher — APPROVED** [review]

**Comment by AlexanderSher** [comment]
LGTM

**Pull request merged**